api/envoy/extensions/filters/udp/udp_proxy/v3/udp_proxy.proto

```diff
@@ -88,4 +89,9 @@ message UdpProxyConfig {
   // The default if not specified is false, that means each data chunk is forwarded
   // to upstream host selected on first chunk receival for that "session" (identified by source IP/port and local IP/port).
   bool use_per_packet_load_balancing = 7;
+
+  // Configuration for :ref:`access logs <arch_overview_access_logs>`
+  // emitted by the this udp_proxy.
+  // repeated config.accesslog.v3.AccessLog access_log = 8;
+  repeated string access_log = 8;
```

Comment:
I think using the `AccessLog` type supports more use-cases. Why did you prefer the `string` type instead?

Reply:
> I think using the `AccessLog` type supports more use-cases. Why did you prefer the `string` type instead?

I find that access logger Instance class is designed to:

```
virtual void log(const Http::RequestHeaderMap* request_headers,
                   const Http::ResponseHeaderMap* response_headers,
                   const Http::ResponseTrailerMap* response_trailers,
                   const StreamInfo::StreamInfo& stream_info) PURE;
```
There is a reference type at the end of the parameter list, so this interface may not be reused by directly inheriting.

Reply:
You are probably going to want to synthesize a StreamInfo anyway, since many of the fields will apply and you can document which ones don't (see the existing access log documentation for TCP).

Reply:
> You are probably going to want to synthesize a StreamInfo anyway, since many of the fields will apply and you can document which ones don't (see the existing access log documentation for TCP).

Yeah, I did some statistics, the existing fileds which may be reused for udp access logging in StreamInfo:

```
1. %START_TIME%
2. %BYTES_RECEIVED%: Downstream bytes received on connection.
3. %BYTES_SENT%: Downstream bytes sent on connection.
```
It looks a bit less. Do you think it is necessary to create new StreamInfo class, like UdpStreamInfo inherited from StreamInfo?

Reply:
> You can probably just reuse the existing StreamInfoImpl. The main thing is that all of the substitution formatter code will expect a StreamInfo. So to make it "just work" you will need an instance of that interface.

OK, I will reuse uint64_t variables in StreamInfoImpl, thanks.